plugin/pkg/admission/priority/admission.go

```diff
@@ -231,12 +231,30 @@ func (p *PriorityPlugin) getDefaultPriorityClass() (*scheduling.PriorityClass, e
 	if err != nil {
 		return nil, err
 	}
+	// In case more than one global default priority class is added as a result of a race condition,
+	// we pick the one with the lowest priority value.
+	var defaultPCs []*scheduling.PriorityClass
+	var defaultPC *scheduling.PriorityClass
 	for _, pci := range list {
 		if pci.GlobalDefault {
-			return pci, nil
+			if defaultPC == nil {
+				// This is the first default PC found.
+				defaultPC = pci
+			} else {
+				defaultPCs = append(defaultPCs, defaultPC, pci)
+			}
+		}
+	}
+	if len(defaultPCs) > 0 {
```

Comment:
Should be able to do the comparison in place rather than accumulating

Reply:
Absolutely. Fixing a PR while running to a meeting never works. Done.